### PR TITLE
G0 5539: Advance receipt adjustment uncheck then pop up verify message 

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html [hunk 1]
```diff
@@ -1129,3 +1129,12 @@ <h3 *ngIf="!openDropDown" (dblclick)="openHeaderDropDown()" class="d-inline-bloc
     </div>
 </ng-template>
 <!-- Tourist scheme template block end-->
+<!--Advance receipt adjusted remove confirmation modal -->
+<div bsModal #advanceReceiptRemoveConfirmationModal="bs-modal" class="modal fade" role="dialog" [config]="{keyboard:true}" tabindex="-1">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html:1133]: `delete-template-confirmation-modal` should not be in the DOM if the modal is not shown consider adding `*ngIf` to the component with condition `advanceReceiptRemoveConfirmationModal.isShown`

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 2]
```diff
@@ -86,6 +86,7 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
     @ViewChild('tax') public taxControll: TaxControlComponent;
     @ViewChild('updateBaseAccount') public updateBaseAccount: ModalDirective;
     @ViewChild(BsDatepickerDirective) public datepickers: BsDatepickerDirective;
+    @ViewChild('advanceReceiptRemoveConfirmationModal') public advanceReceiptRemoveConfirmationModal: ModalDirective;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:89]: Do add JS doc

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 3]
```diff
@@ -177,10 +178,12 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
     /** True, if company country supports other tax (TCS/TDS) */
     public isTcsTdsApplicable: boolean;
-
     /** True, if all the transactions are of type 'Tax' or 'Reverse Charge' */
     private taxOnlyTransactions: boolean;
-
+    /** Remove Advance receipt confirmation flag */
+    public confirmationFlag: string = '';
+    /** Remove Advance receipt confirmation message */
+    public removeAdvanceReceiptConfirmationMessage: string = 'If you change the type of this receipt, all the related advance receipt adjustments in invoices will be removed. Are you sure you want to proceed';
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:186]: Message contains a question without question mark (?). Please check

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 4]
```diff
@@ -1505,4 +1513,21 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
         }
     }
+    /**
+     * Advance receipt adjustment remove model action response
+     *
+     * @param {*} userResponse  Action message
+     * @memberof UpdateLedgerEntryPanelComponent
+     */
+    public onAdvanceReceiptRemoveCloseConfirmationModal(userResponse: any): void {
+        if (userResponse.response) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1523]: This method can be made short as:
`
if (userResponse) {
this.isAdvanceReceipt = !userResponse.response;
this.handleAdvanceReceiptChange();
this.advanceReceiptRemoveConfirmationModal.hide();
}
`

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 5]
```diff
@@ -1278,6 +1282,11 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
         if (this.shouldShowAdvanceReceiptMandatoryFields) {
             this.vm.generatePanelAmount();
         }
+        if (!this.isAdvanceReceipt) {
+            if (this.isAdjustedInvoicesWithAdvanceReceipt && this.vm.selectedLedger && this.vm.selectedLedger.voucherGeneratedType === 'receipt') {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1286]: Please use ENUM here for receipt

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 6]
```diff
@@ -1505,4 +1514,17 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
+    /**
+     * Advance receipt adjustment remove model action response
+     *
+     * @param {*} userResponse  Action message
+     * @memberof UpdateLedgerEntryPanelComponent
+     */
+    public onAdvanceReceiptRemoveCloseConfirmationModal(userResponse: any): void {
+        if (userResponse) {
+             this.isAdvanceReceipt = !userResponse.response;
+            this.handleAdvanceReceiptChange();
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1526]: code is not formatted

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts (hunk 5 above)
Comment (rohit-walkover) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1286]: Which ENUM have to use here

**Pull request merged**